### PR TITLE
CB-16621: Added E2E test for resize recovery from a Datalake Restore failure.

### PR DESCRIPTION
**Pull request by sxxgrc**
Jira: https://jira.cloudera.com/browse/CB-16621

Objective: Add e2e testing for recovery from general resize failures. This change adds the required changes for these tests as well as the specific data restore failure recovery test.

I have verified that the few methods I changed outside of the integration test directory are only called by the integration test logic.

Part of my changes also include adding both the `SdxResizeTests` and `SdxResizeRecoveryTests` to the AWS longrunning (just recovery for this one) and Azure longrunning testsuites as we would like to ensure they are continuously verified for both.

I have tested this locally and ensured that everything is working as desired.

### CODE REVIEW

**Comment by sxxgrc** [comment]
I have labeled this as **DO NOT MERGE** as these changes depend on other recently merged changes which are currently breaking the e2e tests.

Until I have remedied this I do not want to merge in further changes.

**Review by RokolyA94 — APPROVED** [review]

**Comment by sxxgrc** [comment]
We no longer really care about the DR service in the E2E stuff so this can be merged in once I remove those checks and fix the PR.

**Pull request closed without merging**
